src/librustc_typeck/check/pat.rs

```diff
@@ -1414,7 +1415,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else if let Some(pat_len) = len.checked_sub(min_len) {
                 // The variable-length pattern was there,
                 // so it has an array type with the remaining elements left as its size...
-                return (self.tcx.mk_array(element_ty, pat_len), arr_ty);
+                return (Some(self.tcx.mk_array(element_ty, pat_len)), arr_ty);
             } else {
                 // ...however, in this case, there were no remaining elements.
                 // That is, the slice pattern requires more than the array type offers.
```

Comment:
`slice.is_some()` but `slice_ty == None` holds here ==> ICE. (If there is no ICE then we have a lack of tests?)

Reply:
Sorry, I think I'm misunderstanding? How can this ever happen here given that we directly return `Some(slice_ty)`?

Reply:
This has `return (None, updated_arr_ty)` but `slice.is_some()` holds.

I suspect this function should be inverted a bit to check `slice.is_none()` first; that might make reasoning simpler, but that's for another PR.